### PR TITLE
Docs: Improve example of Lint/ToJSON

### PR DESCRIPTION
**Pull request by jaredbeck**
I'm trying to understand the existing description of this cop. Does this new example improve the description?

-----------------

* [x] Wrote [good commit messages][1].
* [N/A] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [N/A] Added tests.
* [N/A] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [N/A to this type of code comments?] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by marcandre** [comment]
Good question. Isn't defining `to_json` to be avoided altogether in favor of `as_json`?

**Comment by bbatsov** [comment]
Here's the original ticket that prompted the creation of that cop https://github.com/rubocop-hq/rubocop/issues/6378 Perhaps something has changed in the meantime?

File: lib/rubocop/cop/lint/to_json.rb [hunk 1]
```diff
@@ -9,13 +9,23 @@ module Lint
...
-      #   # bad
-      #   def to_json
-      #   end
+      #   class Point
+      #     attr_reader :x, :y
       #
-      #   # good
-      #   def to_json(*_args)
-      #   end
+      #     # bad
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/to_json.rb:15]: Don't forget to regen the docs.

**Comment by jaredbeck** [comment]
Docs regenerated

**Comment by bbatsov** [comment]
Thanks!

**Comment by akostadinov** [comment]
> Good question. Isn't defining to_json to be avoided altogether in favor of as_json?

Not for non-rails projects.

**Pull request merged**